webscraping.qmd

```diff
@@ -473,22 +479,22 @@ This includes a few empty columns, but overall does a good job of capturing the
 However, we need to do some more processing to make it easier to use.
 First, we'll rename the columns to be easier to work with, and remove the extraneous whitespace in rank and title.
 We will do this with `select()` (instead of `rename()`) to do the renaming and selecting of just these two columns in one step.
-Then, we'll apply `separate_wider_regex()` (from @sec-extract-variables) to pull out the title, year, and rank into their own variables.
+Then we'll remove the new lines and extra spaces then apply `separate_wider_regex()` (from @sec-extract-variables) to pull out the title, year, and rank into their own variables.
 
 ```{r}
-ratings <- table |> 
+ratings <- table |>
   select(
     rank_title_year = `Rank & Title`,
     rating = `IMDb Rating`
   ) |> 
   mutate(
-    rank_title_year = str_squish(rank_title_year)
+    rank_title_year = str_replace_all(rank_title_year, "\n +", " ")
```

Comment:
Does `str_squish()` not do the job here?

Reply:
Reviewers commented that it needed explanation, which I didn't want to add.